client/src/features/sessionsV2/components/SessionButton/ActiveSessionButton.tsx

```diff
@@ -615,28 +594,26 @@ function ModifySessionModalContent({
           <Col>
             {message}
             <p>
-              <span className="fw-bold me-3">Current resources:</span>
+              <span className={cx("fw-bold", "me-3")}>Current resources:</span>
               <span>
-                <SessionRowResourceRequests
-                  resourceRequests={resources.requests}
-                />
+                <SessionRowResourceRequests resourceRequests={resources} />
               </span>
             </p>
             <div className="field-group">{selector}</div>
           </Col>
         </Row>
       </ModalBody>
       <ModalFooter>
-        {status === "hibernated" && (
+        {state === "hibernated" && (
           <Button
             disabled={
               isLoading ||
               !resourcePools ||
               resourcePools.length == 0 ||
               isError ||
               currentSessionClass == null ||
-              (currentSessionClassId != null &&
-                currentSessionClassId === currentSessionClass?.id)
+              (resource_class_id != null &&
+                resource_class_id === `${currentSessionClass?.id}`)
```

Comment:
Since this logic is used in a couple of places:

```suggestion
resourceClassMatchesCurrent(resource_class_id, currentSessionClass)
```

```
function resourceClassMatchesCurrent(resourceClassId: string | undefined, currentSessionClass: string) {
    return (resourceClassId != null && resourceClassId === ""+currentSessionClass.id)
}
```

Reply:
why not just?
```
resource_class_id === `${currentSessionClass?.id}
```

This is all that's needed here (`resource_class_id` cannot be null-ish and equal a string).